File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/ContactPersonColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ContactPersonRow } from "./ContactPersonRow";

export interface ContactPersonColumns {
    ContactPersonId: Column<ContactPersonRow>;
    PersonFullName: Column<ContactPersonRow>;
}

export class ContactPersonColumns extends ColumnsBase<ContactPersonRow> {
    static readonly columnsKey = 'CrmDB.ContactPerson';
    static readonly Fields = fieldsProxy<ContactPersonColumns>();
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/ContactPerson/ContactPersonColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB.Columns;

[ColumnsScript("CrmDB.ContactPerson")]
[BasedOnRow(typeof(ContactPersonRow), CheckNames = true)]
public class ContactPersonColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int ContactPersonId { get; set; }

    [EditLink]
    public string PersonFullName { get; set; }    
}